### ISSUE TITLE
The Legend of Heroes: Trails of Cold Steel III (991270)

### ISSUE THREAD

**Issue by krauser87** [ValveSoftware/Proton#3683]
# Compatibility Report
- Name of the game with compatibility issues: The Legend of Heroes: Trails of Cold Steel II
- Steam AppID of the game: 991270

## System Information
- GPU: NVIDIA TU106 [GeForce RTX 2060]
- Driver/LLVM version: nvidia v: 435.21 
- Kernel version:  5.3.0-42-generic
- Link to full system information report as [Gist](https://gist.github.com/):https://gist.github.com/krauser87/50adce6ad8d383463e2b8e673228b1f9
- Proton version: 5.0-5

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [ ] that I have checked whether there are updates for my system available.

[steam-991270.log.tar.gz](https://github.com/ValveSoftware/Proton/files/4374145/steam-991270.log.tar.gz)

## Symptoms 
In low settings or max settings, with vsync enabled or disabled, the game always crashes(or should I say exits without any message, to the desktop/steam page after 15-30m of gameplay). I've tried with different proton versions, from the most recent one to almost the oldest, the results are the same. No additional tweaks were done to the game since everything works(besides the crashes), including the videos.

**Comment by kisak-valve** [comment]
Hello @krauser87, in general Proton logs compress well. Please consider throwing it in an archive.

**Comment by krauser87** [comment]
OP post updated with the proton log file compressed in tar.gz

**Comment by doitsujin** [comment]
> eventfd: Too many open files

Can you try setting your Steam launch options to `PROTON_NO_ESYNC=1 %command%`?

**Comment by krauser87** [comment]
> > eventfd: Too many open files
> 
> Can you try setting your Steam launch options to `PROTON_NO_ESYNC=1 %command%`?

Just did and for now no crashes at all. Will keep playing and see if the issue is gone for good.
Edit: The crashes have stopped. There's some minor stuttering(very minor) compared to playing it on windows, aside from that, game is working perfectly.

**Comment by TheFynx** [comment]
`PROTON_NO_ESYNC=1 %command%` has fixed my crashing issues as well. 

Providing system info for those curious
- GPU: GeForce RTX 2070 SUPER
- Driver/LLVM version: 4.6.0 NVIDIA 440.64
- Kernel Version: 5.5.10-arch1-1
- System Info: <https://gist.github.com/TheFynx/d0866bc7dbd87642caf5841229ebe645>
- Proton version: 5.0-5

It does freeze/crash when trying to exit the game (for me at least). Though since I want the game to close and the game is saved by the time I get to the exit on the title screen... kill it with :fire:

**Comment by webgeek1234** [comment]
I can also confirm that disabling esync fixes the crashes for me on Fedora 31. I get the freeze on exit as well, but doing a force close from the menu 'works', but is certainly not ideal.

One other thing I've noticed. Clipping on overlays doesn't seem to always work. For example, in battle, the hp overlays will go on top of the menus. This hasn't happened all the time for me. It seems to be consistent within a single execution of the game: layering is all correct or layering is consistently broken.

**Comment by doitsujin** [comment]
> For example, in battle, the hp overlays will go on top of the menus.

You sure this isn't just a game bug?

Whether UI element is drawn on top of another solely depends on the order in which the game issues the respective draw tcalls, this isn't something that Proton could possibly be messing up (assuming I understand the problem correctly). Also not sure if I've seen this so far.

**Comment by webgeek1234** [comment]
It's possible. I don't have windows to test it there. I did not see such issues on the PS4 version. Fwiw, I have not seen this in my last few runs. It may have been something related to the esync issue throwing the engine out of whack.

**Comment by BillFleming** [comment]
I set my FD limit to something crazy high (8388608) and played the game for several days with no crashes, then switched to fysnc and still no crashes. With the latest custom proton builds with video patches the videos also now play in the game. I finished the game and got almost all achievements and completed all quests (missed a chest somewhere lolz). Did not experience any issues or missing features. So once video patches land in normal proton this game should be whitelisted. Also the videos play now in the other cold steel games. :-)
Note that for this game you need to select the main screen by pressing a key/button then wait for the intro video to play, to confirm that video playback is working. Near the very very end of the game it will play several videos and these all worked without issue also. The game made it through all the videos, dialog, and the cliffhanger ending without any crashes or anything.

**Comment by mcmurphy8097234789** [comment]
Wanted to add update to this

**Compatibility Report**
Name of the game with compatibility issues: The Legend of Heroes: Trails of Cold Steel II
Steam AppID of the game: 991270

**System Information**

> OS: Pop!_OS 21.04
>KERNEL: 5.11.0-7620-generic
>CPU: Intel Core i7-3630QM @ 2.40GHz
>GPU: NVIDIA NVIDIA GeForce GTX  680M
>GPU DRIVER: NVIDIA 470.57.02
>RAM: 12 GB 
>Proton version: 6.3-6

**I confirm:**

[x] that I haven't found an existing compatibility report for this game.
[x] that I have checked whether there are updates for my system available.

**Symptoms**
Game crashes after about 15-20 minutes. Disabling esync in the launch options seems to fix this problem. 
Game also is unresponsive to wired xbox 360 controller, whereas Proton 6.15-GE-2 is able to interface with controller without issues or need for user configuration

**Comment by RAMChYLD** [comment]
> It does freeze/crash when trying to exit the game (for me at least). Though since I want the game to close and the game is saved by the time I get to the exit on the title screen... kill it with

I'm also experiencing the same issue where the game freezes when I select exit. In fullscreen mode this is especially bad since it keeps mouse locked and I can't click on the button to end task when the frozen task popup appears. I only can free myself by ctrl-alt-F2 to another VT and do a killall -9 from there. 

System Information:
- GPU: AMD Radeon Vega 56
- Driver/LLVM version: `21.3~git2108310600.213548~oibaf~f/12.0.1-1~oibaf~f`
- Kernel version: Linux 5.11.0-27-lowlatency
- Link to full system information report as Gist: https://gist.github.com/RAMChYLD/8bd21ed8b8387d53134beba9c2e8912e
- Proton version: Experimental

**Comment by zany130** [comment]
alt-tabbing causes the game to freeze

tried a fresh prefix and game no longer launches
[steam-991270.log](https://github.com/ValveSoftware/Proton/files/8211416/steam-991270.log)
nvm
seems to be caused by mangohud disabling allows it to work normally (can alt-tab and play normally game will still frezzse the if you alt-f4 or try to exit the game from the in-game menu

crash on startup of the game
[steam-991270.log](https://github.com/ValveSoftware/Proton/files/8227180/steam-991270.log)
started  using 6.3-8 as it would randomly not launch  on 7.0-2

**Comment by zany130** [comment]
**Sys Info**
```
inxi -b
System:
  Host: Kiseki Kernel: 5.17.5-256-tkg-pds arch: x86_64 bits: 64
    Desktop: KDE Plasma v: 5.24.5 Distro: Garuda Linux
Machine:
  Type: Desktop Mobo: ASRock model: X470 Taichi serial: <superuser required>
    UEFI: American Megatrends v: P4.86 date: 01/27/2022
CPU:
  Info: 6-core AMD Ryzen 5 2600X [MT MCP] speed (MHz): avg: 4093
    min/max: 2200/3600
Graphics:
  Device-1: NVIDIA GP104 [GeForce GTX 1080] driver: nvidia v: 510.68.02
  Display: x11 server: X.Org v: 21.1.3 with: Xwayland v: 22.1.1 driver: X:
    loaded: nvidia gpu: nvidia,nvidia-nvswitch resolution: 1: 2560x1080
    2: 3840x2160~60Hz
  OpenGL: renderer: NVIDIA GeForce GTX 1080/PCIe/SSE2
    v: 4.6.0 NVIDIA 510.68.02
Network:
  Device-1: Intel I211 Gigabit Network driver: igb
Drives:
  Local Storage: total: 2.96 TiB used: 2.97 TiB (100.4%)
Info:
  Processes: 467 Uptime: 4h 54m Memory: 31.27 GiB used: 18.75 GiB (60.0%)
  Shell: fish inxi: 3.3.15
  ```

gist to full steam system info https://gist.github.com/zany130/b011ecd97d8c6b6662ecc0e3081f4b0d
___
**Issue**
Game crashes after a few hours of playtime ( sometimes it will crash with the game built in crash reporter getting executed or times graphics will just hang but audio will continue and you can hear the  character moving. (some times it will even start to show previously shown graphics (parts of menus or other levels as if it where dumping the memory or cache)

___
**Logs**
https://u.pcloud.link/publink/show?code=XZRFlpVZwLpbNrFFzXBWWFEH63iqbFItFbEX

___
**What I have tried**

* different proton versions I normally use the latest GE-Proton for FSR support.
* disabling extra components I am using such as reshade, steamtinkerlaunch, and mangohud
* for mangohud see here https://github.com/flightlessmango/MangoHud/issues/743 (TLDR I started having the issue after a mangohud update so I tried disabling mangohud this seems to help with how long I can play without crashing (from  30 -60 minutes with mangohud enabled on newer versions to 3+ hours on older versions and with mangohud disabled)
* Different kernel versions (arch vanilla kernel and arch lts kernel)

**Comment by madewokherd** [comment]
This looks like it might be relevant:
```
err:   DxvkSubmissionQueue: Command submission failed: VK_ERROR_DEVICE_LOST
```

**Comment by zany130** [comment]
yeah it may have been a driver issue. upgraded to a 6700xt and I don't get that anymore or any crashes for that matter
